### PR TITLE
Add Target.ObjectFormat, use it

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

File: src/dmd/dsymbolsem.d [hunk 1]
```diff
@@ -1636,7 +1636,7 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
 
         // Should be merged with PragmaStatement
         //printf("\tPragmaDeclaration::semantic '%s'\n", pd.toChars());
-        if (target.mscoff)
+        if (target.objectFormat() == Target.ObjectFormat.coff)
```
Comment (ibuclaw) [inline src/dmd/dsymbolsem.d:1639]: This condition should be a target hook.

File: src/dmd/dsymbolsem.d [hunk 2]
```diff
@@ -3459,7 +3459,8 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
 
                 /* These quirky conditions mimic what VC++ appears to do
                  */
-                if (target.mscoff && cd.classKind == ClassKind.cpp &&
+                if (target.objectFormat() == Target.ObjectFormat.coff &&
+                    cd.classKind == ClassKind.cpp &&
```
Comment (ibuclaw) [inline src/dmd/dsymbolsem.d:3463]: And this one too.

File: src/dmd/target.d [hunk 3]
```diff
@@ -285,6 +293,21 @@ extern (C++) struct Target
             assert(0, "unknown environment");
     }
 
+    /**
+     Determine the object format to be used
+     */
+    Target.ObjectFormat objectFormat()
```
Comment (ibuclaw) [inline src/dmd/target.d:299]: Then this can be extern(D) instead.

**Comment by thewilsonator** [comment]
Feel free to do those in a follow up PR.

**Review by ibuclaw — DISMISSED** [review]
#13513

**Comment by RazvanN7** [comment]
ping @thewilsonator

**Comment by thewilsonator** [comment]
pong @ibuclaw

**Review by RazvanN7 — APPROVED** [review]

**Review by maxhaton — APPROVED** [review]

**Pull request merged**